api/v2/checluster_types.go

```diff
@@ -0,0 +1,637 @@
+	// PostgreSQL Database hostname that the Che server uses to connect to.
+	// Override this value ONLY when using an external database. See field `externalDb`.
+	// +kubebuilder:default:="postgres"
+	// +optional
+	PostgresHostName string `json:"postgresHostName,omitempty"`
+	// PostgreSQL Database port that the Che server uses to connect to. Defaults to 5432.
+	// Override this value ONLY when using an external database. See field `externalDb`. In the default case it will be automatically set by the Operator.
+	// +optional
+	// +kubebuilder:default:="5432"
+	PostgresPort string `json:"postgresPort,omitempty"`
+	// PostgreSQL database name that the Che server uses to connect to the DB.
+	// +optional
+	// +kubebuilder:default:="dbche"
+	PostgresDb string `json:"postgresDb,omitempty"`
+	// The secret that contains PostgreSQL `user` and `password` that the Che server uses to connect to the DB.
+	// The secret must have `app.kubernetes.io/part-of=che.eclipse.org` label.
+	// +optional
+	// +kubebuilder:default:="postgres-credentials"
+	CredentialsSecretName string `json:"credentialsSecretName,omitempty"`
+	// PVC settings for PostgreSQL database.
+	// +optional
+	Pvc PVC `json:"pvc,omitempty"`
+}
+
+// Che server metrics configuration
+type ServerMetrics struct {
+	// Enables `metrics` the Che server endpoint.
+	// +kubebuilder:default:=true
+	// +optional
+	Enable bool `json:"enable"`
+}
+
+// Configuration settings for installation and configuration of the Kubernetes Image Puller
+// See https://github.com/che-incubator/kubernetes-image-puller-operator
+// +k8s:openapi-gen=true
+type ImagePuller struct {
+	// Install and configure the Community Supported Kubernetes Image Puller Operator. When set to `true` and no spec is provided,
+	// it will create a default KubernetesImagePuller object to be managed by the Operator.
+	// When set to `false`, the KubernetesImagePuller object will be deleted, and the Operator will be uninstalled,
+	// regardless of whether a spec is provided.
+	// If the `spec.images` field is empty, a set of recommended workspace-related images will be automatically detected and
+	// pre-pulled after installation.
+	// Note that while this Operator and its behavior is community-supported, its payload may be commercially-supported
+	// for pulling commercially-supported images.
+	Enable bool `json:"enable"`
+	// A KubernetesImagePullerSpec to configure the image puller in the CheCluster
+	// +optional
+	Spec imagepullerv1alpha1.KubernetesImagePullerSpec `json:"spec"`
+}
+
+// Settings for installation and configuration of the DevWorkspace operator
+// See https://github.com/devfile/devworkspace-operator
+// +k8s:openapi-gen=true
+type DevWorkspace struct {
+	// Deployment override options.
+	// +optional
+	Deployment Deployment `json:"deployment,omitempty"`
+	// Maximum number of the running workspaces per user.
+	// +optional
+	RunningLimit string `json:"runningLimit,omitempty"`
```

Comment:
This might make more sense as a dashboard option (rather than DevWorkspace). Currently, the dashboard is what imposes the limit on running workspaces, and DWO does not have functionality to do this.